app/src/main/java/com/mybus/dao/RealmDao.java

```diff
@@ -0,0 +1,176 @@
+package com.mybus.dao;
+
+import android.content.Context;
+
+import com.mybus.model.FavoriteLocation;
+import com.mybus.model.UsageTrackable;
+
+import java.util.Date;
+
+import io.realm.Realm;
+import io.realm.RealmConfiguration;
+import io.realm.RealmObject;
+import io.realm.RealmResults;
+
+/**
+ * Created by Julian Gonzalez <jgonzalez@devspark.com>
+ */
+public abstract class RealmDao<T extends RealmObject> {
+    private Realm mRealm;
+    private Class<T> mType;
+
+    protected RealmDao(Context c, Class<T> clazz) {
+        RealmConfiguration realConfig = new RealmConfiguration.Builder(c)
+                .deleteRealmIfMigrationNeeded()
+                .build();
+        this.mRealm = Realm.getInstance(realConfig);
+        this.mType = clazz;
+    }
+
+    /**
+     * Save or update the given item depends if it already exists or not.
+     * @param item
+     * @return The item updated or null if an error occurred
+     */
+    public T saveOrUpdate(T item) {
+        if (item == null) {
+            return null;
+        }
+        try {
+            mRealm.beginTransaction();
+            T updated = mRealm.copyToRealmOrUpdate(item);
+            mRealm.commitTransaction();
+            return updated;
+        } catch (IllegalArgumentException e) {
+            return null;
```

Comment:
Why would you get an IllegalArgumentException here and why would you swallow it and return null